### ISSUE TITLE
Error unliking posts

### ISSUE THREAD

**Issue by mh10001** [instabot-py/instabot.py#2178]
Check this or we will delete your issue. (fill in the checkbox with an X like so: [x])
- [x] I have searched for other issues with the same problem or similar feature requests. 

#### Select one:
- [x ] Bug
- [ ] Feature Request
- [ ] Technical Help
- [ ] Other

#### Environment
- [ ] Are you hosting the bot on a VPS or other Cloud hosting?
- [ ] If so, have you tried running the code on your local machine?

#### Operating System? (include version)
- [ ] macOS
- [x ] Windows
- [ ] Linux (include flavour)
- [ ] Android (include terminal)

#### Python Version Requirement
- [x ] I am using Python 3.6 or above

#### Exact Python Version?
```
>python --version
Python 3.7.3

```

#### Pip Version?
```
>pip3 --version
pip 19.1.1 from c:\users\XXXXXX\appdata\local\programs\python\python37\lib\site-packages\pip (python 3.7)

```

#### Description of your issue

```
Every time the script attempts to unlike a photo, the script prints error: 

24.05.2019_17:10 Trying to unlike media
24.05.2019_17:11 Couldn't unlike media, resuming.

NOTE: Debugging shows that the Instagram HTTP server is responding with an HTTP Error 405 every time the unlike URL is requested by the script.  All other features work- liking posts, posting comments, following, unfollowing.

```

#### Config

```
[DEFAULT]
username = none
password = none
like_per_day = 209
media_min_like = 5
media_max_like = 0
comments_per_day = 30
max_like_for_one_tag = 46
follow_per_day = 80
follow_time = 36000
unfollow_per_day = 107
unfollow_break_min = 1
unfollow_break_max = 10
log_mod = 0
proxy = 
unfollow_selebgram = False
unfollow_probably_fake = False
unfollow_inactive = False
unfollow_recent_feed = False
comment_list = [[""]]
tag_list = ["follow4follow", "f4f", "cute", "l:212999109"]
tag_blacklist = ["rain", "thunderstorm", "fuck", "naked", "nsfw"]
unwanted_username_list = ["stuff", "art", "project", "love", "life", "food", "blog", "free", "keren", "photo", "graphy", "indo", "travel", "art", "shop", "store", "sex", "toko", "jual", "online", "murah", "jam", "kaos", "case", "baju", "fashion", "corp", "tas", "butik", "grosir", "karpet", "sosis", "salon", "skin", "care", "cloth", "tech", "rental", "kamera", "beauty", "express", "kredit", "collection", "impor", "preloved", "follow", "follower", "gain", ".id", "_id", "bags"]
unfollow_whitelist = ["example_user_1", "example_user_2"]

[XXXXX]
username = XXXXX
password = XXXXX
start_at_h = 12
start_at_m = 24
end_at_h = 18
end_at_m = 45
like_per_day = 100
unlike_per_day = 50
time_till_unlike = 18000
comments_per_day = 0
max_like_for_one_tag = 50
follow_per_day = 200
follow_time = 200000
follow_time_enabled = True
user_min_follow = 100
user_max_follow = 11000
unfollow_not_following = True
unfollow_per_day = 20
tag_list = ["gaybeard", "gaymuscles", "gaytravel"]
unfollow_selebgram = False
unfollow_probably_fake = False
unfollow_inactive = False
unfollow_recent_feed = False




```

**Comment by KiillThemAll** [comment]
Same problem.

**Comment by nickgryg** [comment]
Same here:
2019-06-03 14:35:38,293 - InstaBot - DEBUG - Trying to unlike media
2019-06-03 14:35:38,406 - InstaBot - CRITICAL - Couldn't unlike media, resuming.

**Comment by saabia** [comment]
this never worked by me

**Comment by dmgl** [comment]
Any news?

**Comment by nickgryg** [comment]
I've made some debugging here. 

I've added more debug information to auto_unlike() function:
``` 
   def auto_unlike(self):
        checking = True
        while checking:
            media_to_unlike = self.persistence.get_medias_to_unlike()
            if media_to_unlike:
                request = self.unlike(media_to_unlike)
                if request.status_code == 200:
                    self.persistence.update_media_complete(media_to_unlike)
                else:
                    self.logger.critical(f"Couldn't unlike {media_to_unlike} - https://www.{self.get_instagram_url_from_media_id(media_to_unlike)} media, resuming.")
                    self.logger.debug(f"Response message: {request.text}; Response code: {request.status_code}")
                    checking = False
            else:
                self.logger.debug("no medias to unlike")
                checking = False
```

The output is:
```
2019-06-12 11:56:17,622 - InstaBot - CRITICAL - Couldn't unlike 2058927438624XXXXXX - https://www.instagram.com/p/XXXXXXXXXXX/ media, resuming.
2019-06-12 11:56:17,622 - InstaBot - DEBUG - Response message: missing media; Response code: 400
```

And this is totally true. If you go to that link, you will see that the page is not exist anymore. The strange thing that it is happening to every media which was liked before. Seems like instagram changes media's id regularly after some time. Currently I have `time_till_unlike = 86400`.

**Comment by nickgryg** [comment]
I've made some debugging here. 

I've added more debug information to auto_unlike() function:
``` 
    def auto_unlike(self):
        checking = True
        while checking:
            media_to_unlike = self.persistence.get_medias_to_unlike()
            if media_to_unlike:
                request = self.unlike(media_to_unlike)
                if request.status_code == 200:
                    self.persistence.update_media_complete(media_to_unlike)
                    self.logger.info(f”Unliked {media_to_unlike} - https://www.{self.get_instagram_url_from_media_id(media_to_unlike)} media")
                    self.logger.debug(f"Response message: {request.text}; Response code: {request.status_code}")
                else:
                    self.logger.critical(f"Couldn't unlike {media_to_unlike} - https://www.{self.get_instagram_url_from_media_id(media_to_unlike)} media, resuming.")
                    self.logger.debug(f"Response message: {request.text}; Response code: {request.status_code}")
                    checking = False
            else:
                self.logger.debug("no medias to unlike")
                checking = False
```

The output is:
```
2019-06-12 12:28:50,474 - InstaBot - INFO - Unliked 2062038537068XXXXXX - https://www.instagram.com/p/XXXXXXXXXXX/ media
2019-06-12 12:28:50,474 - InstaBot - DEBUG - Response message: {"status": "ok"}; Response code: 200
2019-06-12 11:56:17,622 - InstaBot - CRITICAL - Couldn't unlike 2058927438624XXXXXX - https://www.instagram.com/p/XXXXXXXXXXX/ media, resuming.
2019-06-12 11:56:17,622 - InstaBot - DEBUG - Response message: missing media; Response code: 400
```

And this means that unliking works as expected. We just do not see logs about successful unlikings and see only unsuccessful ones. It is true, If you go to the unsuccessful unliking link, you will see that the page is not exist anymore. It could be that the media was deleted or replaced.

It seems it is obvious if the media is not exist anymore we don’t need to unlike it.

I’m going to raise a PR to add more debug information to output and to delete deleted from instagram medias from database to not process them in a future.

**Comment by ilMagnifico** [comment]
Same issue

**Comment by nickgryg** [comment]
Issue was fixed in PR #2221
